packages/token-bridge-sdk/src/withdrawals/fetchTokenWithdrawalsFromSubgraph.ts

```diff
@@ -68,18 +68,33 @@ export async function fetchTokenWithdrawalsFromSubgraph({
   })
 
   const result: FetchTokenWithdrawalsFromSubgraphResult[] = await Promise.all(
-    queryResult.data.gatewayWithdrawalDatas.map(async data => {
-      const { l2TxHash, amount: amountStringified } = data
-      const amount = BigNumber.from(amountStringified)
+    queryResult.data.gatewayWithdrawalDatas
+      .map(async data => {
+        const { l2TxHash, amount: amountStringified } = data
+        const amount = BigNumber.from(amountStringified)
 
-      const txReceipt = await l2Provider.getTransactionReceipt(l2TxHash)
-      const l2TxReceipt = new L2TransactionReceipt(txReceipt)
+        const txReceipt = await l2Provider.getTransactionReceipt(l2TxHash)
+        const l2TxReceipt = new L2TransactionReceipt(txReceipt)
 
-      const [l2ToL1Event] = l2TxReceipt.getL2ToL1Events()
-      const tokenAddress = utils.hexDataSlice(l2ToL1Event.data, 16, 36)
+        const [l2ToL1Event] = l2TxReceipt.getL2ToL1Events()
+        if (!l2ToL1Event) {
+          return null
+        }
+        const tokenAddress = utils.hexDataSlice(l2ToL1Event.data, 16, 36)
 
-      return { ...l2ToL1Event, l2TxHash, amount, tokenAddress }
-    })
+        return {
+          ...l2ToL1Event,
+          l2TxHash,
+          amount,
+          tokenAddress
+        }
+      })
+      .filter(
+        (
```

Comment:
Just a small thought. Using this approach we are doing double iteration on `queryResult.data.gatewayWithdrawalDatas`. If we want to optimise it, we can probably use `reduce` or `forEach` and add a null check + push the object to output array within a single iteration.

Reply:
Thank you, found a bug while doing so

Reply:
Actually, `forEach` would not work with array of promise. And `reduce` woul lead to weird code as `accumulator` would need to be awaited each loop